### ISSUE TITLE
[CLI] Wandb crashes when trying to launch a pytorch-lightning experiment

### ISSUE THREAD

**Issue by zplizzi** [wandb/client#2821]
I get this error:

```
Problem at: /home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/loggers/wandb.py 193 experiment
Traceback (most recent call last):
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 870, in init
    run = wi.init()
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 442, in init
    backend.ensure_launched()
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/backend/backend.py", line 201, in ensure_launched
    self.wandb_process.start()
  File "/home/user/.conda/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/home/user/.conda/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/home/user/.conda/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/home/user/.conda/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/home/user/.conda/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
wandb: ERROR Abnormal program exit
Traceback (most recent call last):
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 870, in init
    run = wi.init()
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 442, in init
    backend.ensure_launched()
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/backend/backend.py", line 201, in ensure_launched
    self.wandb_process.start()
  File "/home/user/.conda/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/home/user/.conda/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/home/user/.conda/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/home/user/.conda/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/home/user/.conda/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 234, in prepare
    _fixup_main_from_name(data['init_main_from_name'])
  File "/home/user/.conda/lib/python3.9/multiprocessing/spawn.py", line 258, in _fixup_main_from_name
    main_content = runpy.run_module(mod_name,
  File "/home/user/.conda/lib/python3.9/runpy.py", line 210, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/home/user/.conda/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/user/.conda/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/projects/generally_intelligent/standalone/ballworld/quarantine/zack/simone/train2.py", line 16, in <module>
    trainer.fit(model, dm)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 552, in fit
    self._run(model)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 916, in _run
    self._pre_dispatch()
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 945, in _pre_dispatch
    self._log_hyperparams()
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 972, in _log_hyperparams
    self.logger.log_hyperparams(hparams_initial)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/utilities/distributed.py", line 48, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/loggers/wandb.py", line 210, in log_hyperparams
    self.experiment.config.update(params, allow_val_change=True)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/loggers/base.py", line 42, in experiment
    return get_experiment() or DummyExperiment()
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/utilities/distributed.py", line 48, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/loggers/base.py", line 40, in get_experiment
    return fn(self)
  File "/home/user/.conda/lib/python3.9/site-packages/pytorch_lightning/loggers/wandb.py", line 193, in experiment
    self._experiment = wandb.init(**self._wandb_init) if wandb.run is None else wandb.run
  File "/home/user/.conda/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 908, in init
    six.raise_from(Exception("problem"), error_seen)
  File "<string>", line 3, in raise_from
Exception: problem
```

when trying to run this code:

```
from pytorch_lightning.loggers import WandbLogger
import pytorch_lightning as pl
from pl_examples.basic_examples.mnist_datamodule import MNISTDataModule
from pl_examples.basic_examples.simple_image_classifier import LitClassifier

dm = MNISTDataModule(batch_size=32)
model = LitClassifier()

pl_logger = WandbLogger(
    project="zack_simone",
    name="test",
)
pl.seed_everything(1)

trainer = pl.Trainer(max_epochs=2, gpus=1, logger=pl_logger)
trainer.fit(model, dm)
```

using the current newest versions of wandb (0.12.5) and pytorch lighting ('1.4.9'), on pytorch 1.8.1.

**Comment by kptkin** [comment]
@zplizzi glad to hear it worked! 
Thanks for bringing this issue to our attention, we are aware of it in the regular flow and working on a fix. 
Are you seeing it with the new `service` feature as well? The new flow should resolve many of these issues.

**Comment by zplizzi** [comment]
Yes, definitely even with `wandb.require(experiment="service")`, the experiment frequently does not shut down cleanly and I have to manually kill the process with a SIGKILL signal. Of course I'm not certain what's hanging - if it's in wandb or elsewhere. But I believe the final program output before hanging is usually something in wandb-land.

**Comment by kptkin** [comment]
@zplizzi Thank you! 
I'm going to open an issue for the shut down issue. 
Closing this issue as the original issue is resolved.

**Comment by zplizzi** [comment]
Hey @vanpelt, I just wanted to check if special changes in `pip install --force git+https://github.com/wandb/pytorch-lightning.git@wandb-service-attach` have been merged into the standard version of pytorch-lightning yet? Or if this custom branch is still the recommended way of using wandb + lightning together?

**Comment by kptkin** [comment]
Hi @zplizzi thanks for your inquiry. 
We were running some tests and are planning to upstream the branch in the coming days. 
For now you should still use the custom branch. We will update it here once it is merged.
